docs/proposal/20210823-gpu-support.md

```diff
@@ -0,0 +1,164 @@
+
+* [Single Controller Multitenancy](#gpu-support)
+  * [Glossary](#glossary)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+  * [Proposal](#proposal)
+    * [User Stories](#user-stories)
+      * [Story 1 - using vGPU](#story-1---using-vgpu)
+      * [Story 2 - GPU Direct implementation](#story-2---gpu-direct-implementation)
+      * [Story 3 - PCI passthrough for single node Customer](#story-3---pci-passthrough-for-single-node-customer)
+    * [Requirements](#requirements)
+      * [Functional Requirements](#functional-requirements)
+      * [Non-Functional Requirements](#non-functional-requirements)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Controller Changes](#controller-changes)
+  * [Additional Details](#additional-details)
+    * [Test Plan](#test-plan)
+    * [Graduation Criteria](#graduation-criteria)
+  * [Implementation History](#implementation-history)
+
+## Glossary
+
+* CAPV - An abbreviation of Cluster API Provider vSphere
+
+## Summary
+
+CAPV currently does not support GPU-based infrastructure.
+
+This proposal outlines ways to provision GPU support in CAPV. The proposed changes will maintain backward compatibility and maintain the existing behavior without any extra user configuration.
+
+## Motivation
+
+CAPV does not currently support GPU-accelerated workloads which have a limited addressable market. Competitively, all of the hyper-scale cloud providers (AWS, Azure, & GCP) offer GPU-accelerated virtualization and Kubernetes platforms. Many of them additionally offer robust vertically integrated AI/ML solutions building on open source technology. GPU support is a long-requested feature. With this capability, users can leverage GPU support.
+
+### Goals
+
+1. To enable GPU Support on Vsphere Kubernetes Cluster.
+2. To allow vGPU support for multi-node clusters.
+3. To allow GPU Direct support for multi-node clusters.
+4. To allow PCI passthrough support for single-node clusters.
+
+## Proposal
+
+### User Stories
+
+#### Story 1 - using vGPU
+
+Stacy works at an IT department of an organization where AI/ML models are run every hour and the entire staff wants to leverage GPU-supported nodes. VGPU's are able to provide close to bare-metal performance. Stacy needs a lot of sharing GPUs.
+vGPU provides a full-fleshed cluster from which a virtual GPU can be requested.
+
+#### Story 2 - GPU Direct implementation
+
+Tony is an Engineer at a big organization with multiple centers across the Globe. Tony needs a big GPU cluster to address the needs of his organization. So they intend to use GPU Direct. GPU direct is technically a pool of GPUs connected by a network card in a kind of peer-to-peer network. This is the latest technology and gives the advantage of shared GPUs from a pool. GPU  direct is the technology that helps in linking multiple GPUs. Tony wants to leverage this technology so that it is easier for him to manage resource allocation.
```

Comment:
vGPU and GPU direct seems to be available by Nvidia. Is this vendor dependent? Are there other providers also providing this tech/feature? 
These questions may help design the CR spec schema in my opinion.

Reply:
This is indeed vendor-dependent.  This proposal is keeping in mind Nvidia GPUs and it's specific operator.  Intel has an entirely different architecture and method. Currently, we are targeting Nvidia.

Reply:
Since this implementation is vendor dependent, we should add a section around saying this implementation is specific to Nvidia.